### PR TITLE
Je delete categories

### LINKED ISSUE #173: Delete a Category - 2
As an admin I would like to be able to delete a Category so that I can remove any that are not needed.

**Given** an admin is vising the list of Categories  
**When** they select the `Delete` option  
**Then** they should be prompted to confirm the deletion  

**Given** the admin wishes to confirm the delete  
**When** they the select the option to confirm  
**Then** the Category should be removed from the system  
**And** the admin should be directed back to the Category list  

**Given** the admin decides not to confirm the delete  
**When** they the select the option to reject confirmation  
**Then** the Category should NOT be removed from the system  
**And** the admin should be directed back to the Category list  

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by skratz17**
# Description
Implementing category deletion in client side. Now a user can delete a category from the Category List. This PR includes the reimplementation of the `ConfirmableDeleteButton` to change its underlying `isDeleting` render state to rendering a react-bootstrap Modal instead of rendering two buttons side-by-side.

*Make sure to run an `npm install` before testing this code!* It newly includes the `react-icons` library.

Fixes #173 
## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Please describe step-by-step the process that a reviewer can follow to adequately test this PR. 
- [ ] Go to the categories view at `/categories` and verify that it appears more or less as the wireframe specifies.
- [ ] Click on the trashcan icon for any category and verify that you see a Modal prompting you to confirm your deletion of the category.
- [ ] Exit out of the modal by either: Clicking "Cancel", clicking out of the modal, or typing `ESC` key, and verify that the modal closes and the category is not deleted.
- [ ] Click on the trashcan icon again, and this time click "Delete". Verify that the modal closes and the category is deleted.
- [ ] Verify that you do not see the Delete or Edit buttons rendering for the "Uncategorized" category, as users should not be able to modify this one.

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/categories/CategoryList.js [hunk 1]
```diff
@@ -29,19 +29,34 @@ export default (props) => {
     <>
       <article>
           {categories.map((c) => (
-            <ListGroup key={c.id}>
-              <ListGroup.Item className="mb-2">
-              <Row>
-                <Col className="mt-2">
-              <Category category={c} key={c.id} />
+            <Row className="align-items-center">
```
Comment (mgjeffries) [inline src/components/categories/CategoryList.js:32]: I'm seeing a react warning that "index.js:1 Warning: Each child in a list should have a unique "key" prop." We should add a key here

Comment (skratz17) [reply src/components/categories/CategoryList.js]: got it whoops that should be updated now

**Review by mgjeffries — APPROVED** [review]
Looks great!

**Review by Michaelr-Tyler — APPROVED** [review]
💯

**Pull request merged**